### PR TITLE
[GLT-3763] Run List Page

### PR DESCRIPTION
**Pull request by mengjenn**
Add page showing all the runs. Runs on this page are searchable by run name. 

As per [GLT-3763](https://jira.oicr.on.ca/browse/GLT-3763).

### CODE REVIEW

File: src/main/java/ca/on/oicr/gsi/dimsum/controller/rest/RunListRestController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  @Autowired
+  private RunListManager runListManager;
+
+  @PostMapping
+  public TableData<Run> query(@RequestBody DataQuery query) {
+    validateDataQuery(query);
+    RunSort sort = parseSort(query, RunSort::getByLabel);
+    boolean descending = parseDescending(query);
+    RunFilter baseFilter = parseBaseRunFilter(query);
```
Comment (djcooke) [inline src/main/java/ca/on/oicr/gsi/dimsum/controller/rest/RunListRestController.java:29]: The run list shouldn't have a base filter, as all runs should always be available on it. The base filter is a non-removable filter that's applied to a page, such as filtering by the selected project for the project details page

File: src/main/resources/templates/layout.html [hunk 2]
```diff
@@ -48,6 +48,8 @@
       </div>
     </div>
     <nav th:unless="${hideNav}" class="text-right">
+      <a th:href="@{/runs}" class="text-green-200 font-bold hover:underline">Runs</a>
+      <span class="text-green-200 font-bold"> | </span>
```
Comment (djcooke) [inline src/main/resources/templates/layout.html:52]: This being green makes it blend in with the links. Maybe use the same grey as the line above it?

File: ts/data/run.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      title: "Run",
+      key: "RUN",
+      type: "text",
+      autocompleteUrl: urls.rest.autocomplete.runNames,
+    },
+  ],
+  generateColumns(data) {
+    return [
+      {
+        title: "Name",
```
Comment (djcooke) [inline ts/data/run.ts:25]: Add `sortType: "text"`

File: ts/data/run.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            makeNameDiv(
+              run.name,
+              urls.miso.run(run.name),
+              urls.dimsum.run(run.name)
+            )
+          );
+        },
+      },
+      {
+        title: "Completion Date",
```
Comment (djcooke) [inline ts/data/run.ts:37]: Add `sortType: "date"`

File: ts/util/urls.ts [hunk 5]
```diff
@@ -32,8 +32,10 @@ export const urls = {
       requisitionNames: `${restBaseUrl}/autocomplete/requisition-names`,
       projectNames: `${restBaseUrl}/autocomplete/project-names`,
       donorNames: `${restBaseUrl}/autocomplete/donor-names`,
+      runNames: `${restBaseUrl}/autocomplete/run-names`,
     },
     notifications: `${restBaseUrl}/notifications`,
+    runList: `${restBaseUrl}/runs`,
```
[the remark this answers is not in the document]
Comment (mengjenn) [reply ts/util/urls.ts:38]: This conflicts with line 24

[the remark this answers is not in the document]
Comment (djcooke) [reply ts/util/urls.ts:38]: Oh, right! Can put under `rest.runs.list` then

File: src/main/java/ca/on/oicr/gsi/dimsum/service/RunListManager.java [hunk 6]
```diff
@@ -26,66 +20,15 @@ public class RunListManager {
...
-          filterMap.put(key, filter.predicate());
-        }
-      }
-      for (Predicate<Run> predicate : filterMap.values()) {
-        stream = stream.filter(predicate);
-      }
-    }
-    return stream;
+  public List<Run> getRuns() {
+    return this.runs;
```
Comment (djcooke) [inline src/main/java/ca/on/oicr/gsi/dimsum/service/RunListManager.java:31]: There's no need to maintain this separate list of runs - can get rid of this class entirely. `CaseService` can get runs from the case data: `caseData.getRunsAndLibraries().stream().map(RunAndLibraries::getRun)`

**Review by djcooke — APPROVED** [review]

**Review by mlaszloffy — APPROVED** [review]

**Pull request merged**